R/validator.R

```diff
@@ -0,0 +1,85 @@
+##
+## Author: Chris Bielow
+##
+
+## Functions to validate an mzQC file or in-memory object or an mzQC JSON string
+
+
+#'
+#' Get a validator for mzQC
+#'
+#' @import jsonvalidate
+#'
+getValidator = function()
+{
+  schema = system.file("schema/mzQC_schema.json", package = "rmzqc", mustWork = TRUE)
+  v = jsonvalidate::json_validator(schema, engine = "ajv") ## we need schema draft-07
+  return(v)
+}
+
+#'
+#' Validates an mzQC document which is present as a file.
+#'
+#' The returned TRUE/FALSE has additional attributes in case of errors.
+#' Use attributes(result) to access them.
+#'
+#' @param filepath A path to a file (e.g. "c:/my.mzQC", or "test.mzQC")
+#' @param verbose Show extra information if validation fails
+#' @return TRUE/FALSE if validation was successful/failed
+#'
+#' @export
+#'
+validateFromFile = function(filepath, verbose = TRUE)
+{
+  data = readLines(filepath)
```

Comment:
check and report if reading was successful would be useful.

Reply:
there is a default error message if `readlines()` fails, but its not very good:
```
> rmzqc::validateFromFile("doesnotExist.mzQC")
Error in file(con, "r") : cannot open the connection
In addition: Warning message:
In file(con, "r") :
 
 Error in file(con, "r") : cannot open the connection
```

I've added a check which now says
```
> validateFromFile("doesnotExist.mzQC")
Error in validateFromFile("doesnotExist.mzQC") : 
  The file 'doesnotExist.mzQC' does not exist (or is not readable).
```

Reply:
👍